File: ignite-nest/src/comment/infrastructure/repository/comment-query.adapter.ts
import { Injectable } from '@nestjs/common';
import { CommentInPort } from '../../domain/port/in/comment-in.port';
import { Repository } from 'typeorm';
import { Comment } from '../../entity/comment.entity';
import { InjectRepository } from '@nestjs/typeorm';

@Injectable()
export class CommentQueryAdapter implements CommentInPort {
  constructor(
    @InjectRepository(Comment)
    private readonly commentRepository: Repository<Comment>,
  ) {}

  async findCommentsByPostId(postId: number): Promise<Comment[]> {
    return await this.commentRepository.find({ where: { postId: postId } });
  }

  async findAllCommentList(): Promise<Comment[]> {
    const commets = await this.commentRepository.find();
    return commets;
  }
}

File: ignite-spring/src/main/kotlin/org/rookedsysc/ignitespring/comment/application/adapter/CommentQueryAdapter.kt
package org.rookedsysc.ignitespring.comment.application.adapter

import org.rookedsysc.ignitespring.comment.domain.port.`in`.CommentInPort
import org.rookedsysc.ignitespring.comment.entity.CommentEntity
import org.rookedsysc.ignitespring.comment.infrastructure.repository.CommentRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Repository

@Repository
class CommentQueryAdapter(
    private val commentRepository: CommentRepository
): CommentInPort {
    override fun findAllCommentList(request: Pageable): List<CommentEntity> {
        val comments: Page<CommentEntity> = commentRepository.findAll(request)
        return comments.content
    }

    override fun findByPostId(postId: Long): List<CommentEntity> {
        return commentRepository.findByPostId(postId)
    }
}
